frontend/src/plugins/impl/FileBrowserPlugin.tsx

```diff
@@ -138,6 +138,25 @@ export const FileBrowser = ({
   const [path, setPath] = useState(initialPath);
   const [selectAllLabel, setSelectAllLabel] = useState("Select all");
   const [isUpdatingPath, setIsUpdatingPath] = useState(false);
+  const [isRestricted, setIsRestricted] = useState(restrictNavigation);
+  const [allowMultiSelect, setAllowMultiSelect] = useState(multiple);
+  const [selectMode, setSelectMode] = useState(selectionMode);
+
+  useEffect(() => {
```

Comment:
could we use `useControllableState`?
or could we write our own hook for this? 

it would be nice to document these useEffects. the react team has started to discourage useEffects to trigger state changes, it might be better to do something like this (untested):

```ts
export const useInternalState<T>(initial: T) {
  const [state, setState] = useState(initial);
  const ref = useRef(initial);

  // Whenever the outside state changes, update the internal state
  if (ref.current !== initial) {
    setState(initial); // react is ok to setState during render
    ref.current = initial;
  }

  return [state, setState];
}
```

Reply:
i just pushed up these changes